modules/core/src/mathfuncs.cpp

```diff
@@ -180,6 +180,42 @@ void magnitude( InputArray src1, InputArray src2, OutputArray dst )
     }
 }
 
+void magnitudeSqr( InputArray src1, InputArray src2, OutputArray dst )
+{
+    CV_INSTRUMENT_REGION();
+
+    int type = src1.type(), depth = src1.depth(), cn = src1.channels();
+    CV_Assert( src1.size() == src2.size() && type == src2.type() && (depth == CV_32F || depth == CV_64F));
```

Comment:
Shouldn't we also check all arrays for continuity? `hal::magnitudeSqr*`  functions work with 1D arrays and do not know about row step.

Reply:
continuity should not be a problem thanks to the NAryMatIterator that split data into continuous "planes" (which happen to be rows in this case)